aesd-char-driver/main.c

```diff
@@ -128,6 +129,74 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
     	*f_pos = 0;
     	return count;
 }
+/* Function	: aesd_ioctl
+ * Purpose	: Perform the IOCTL command if the argument by calling writes data to the kernel and expecting information back.
+ * Parameters	: pointer to the aesd_device, command to verify against and a pointer from which the user space is requesting data to the kernel
+ * Returns	: Whether IOCTL can be performed or not, or errors like memory violation, seg faults and incorrect locking
+ * References	: https://man7.org/linux/man-pages/man2/ioctl.2.html
+ */
+long aesd_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
+{
+    size_t bytes_copied_From_user = 0;
+    int lock_status;
+    loff_t pos;
+    struct aesd_dev *dev = NULL;
+    struct aesd_seekto seekto;
+    dev = filp->private_data;
+    lock_status = mutex_lock_interruptible(&dev->lock);
```

Comment:
@srki3050 see comments below - I think you could reduce the locking interval here.  What is dev->lock locking?